core/src/main/java/org/apache/iceberg/FastAppend.java

```diff
@@ -215,7 +213,7 @@ private List<ManifestFile> writeNewManifests() throws IOException {
     }
 
     if (newManifests == null && !newFiles.isEmpty()) {
-      this.newManifests = writeDataManifests(newFiles, spec);
+      this.newManifests = writeDataManifests(Lists.newArrayList(newFiles), spec);
```

Comment:
What about modifying `writeDataManifests` to accept `Collection` and moving the list creation to `divide`?

Reply:
no objection to that. I'll do the same for `writeDeleteManifests`since we'll need that too in the next PR (https://github.com/apache/iceberg/pull/11254)